src/Artsy/Analytics/useTracking.ts

```diff
@@ -0,0 +1,21 @@
+import { TrackingContext } from "Artsy/Analytics/TrackingContext"
+import { useContext } from "react"
+
+export function useTracking() {
+  const tracking = useContext(TrackingContext)
+
+  if (!tracking) {
+    console.error(
+      "[Artsy/Analytics/useTracking] Error: Attempting to call `useTracking` " +
+        "without a TrackingContext present. Did you forget to wrap the top of " +
+        "your component tree with `provideTracking`?"
+    )
+  }
+
+  // If missing a TrackingContext, return a noop for safety
+  const trackEvent = tracking ? tracking.trackEvent : x => x
+
+  return {
+    trackEvent,
+  }
```

Comment:
Seeing as we only return 1 thing here, can we get rid of the object?

Reply:
I like the idea of returning the object here for a couple of reasons:

- It guarantees consistent naming of `trackEvent`
- It leaves the api open for extension

Reply:
> It guarantees consistent naming of trackEvent

What's the benefit of this?

> It leaves the api open for extension

Isn’t that very often the case and yet we don’t do that prematurely otherwise either? I also don't foresee any obvious future additions, but if there are known ones then please do share.

Reply:
> What's the benefit of this?

Consistent naming reduces cognitive load when reading.

> Isn’t that very often the case and yet we don’t do that prematurely otherwise either? I also don't foresee any obvious future additions, but if there are known ones then please do share.

It's just adding a couple of braces. I think it makes sense to not prematurely abstract things, but it would be academic to call this abstraction, or even indirection. It's another way to name a thing.

It also happens to be the idiomatic way to return things from hooks.

Reply:
> Consistent naming reduces cognitive load when reading.

Good, this satisfies my question enough 👍 

> It also happens to be the idiomatic way to return things from hooks.

Is it? I skimmed the API docs before I asked this and only found array destructuring (e.g. state) or not destructuring (eg context).

Reply:
Mostly referring to custom hooks e.g. https://github.com/beizhedenglong/react-hooks-lib

Here's a bunch more libraries: https://nikgraf.github.io/react-hooks/

It's not uncommon to see hooks that only return one unnamed value, but it tends to be the case that it's valid for those values to have different names depending on the context.

Reply:
Thanks for digging those up! 🙏 

> It's not uncommon to see hooks that only return one unnamed value, but it tends to be the case that it's valid for those values to have different names depending on the context.

I see your point 👍

Reply:
Thanks for the good discussion y'all. Objects -> named exports better than default exports; lock the thing down if there should be consistency enforced across the codebase, as we'd expect here.